lib/brain.js

```diff
@@ -3180,7 +3143,81 @@ Brain.prototype._physicalDispense = function _physicalDispense () {
   emit('billDispenserDispensing')
 
   const notesToDispense = optimizeDispense(notes, this.billDispenser.dispenseLimit)
-  dispenseGenerator(notesToDispense, this.tx, this.config.completedTimeout)
+  const batchAmount = notesToDispense.length
+
+  const batchDispense = ({ dispensed, current, error }) => {
+    // short circuit out if a error is found
+    if (error) {
+      return { dispensed, error }
+    }
+
+    const notes = notesToDispense[current - 1]
```

Comment:
Rename ``current - 1`` to descriptive variable. ``current - 1`` should be checked to ensure it's in range.